### PR TITLE
irmin-pack: refactor Inode.Compress types for clarity

### PR DESCRIPTION
**Pull request by CraigFe**
- rename the cases of `Compress.address` to avoid ambiguity about what a "direct" address is (`Direct` is now `Hash`, and `Indirect` is now `Offset`).

- document `Compress.value_t` and manually format it to highlight the symmetries between different cases.

### CODE REVIEW

File: src/irmin-pack/inode.ml [hunk 1]
```diff
@@ -300,17 +300,10 @@ struct
...
-      variant "Compress.address" (fun i d -> function
-        | Indirect x -> i x | Direct x -> d x)
-      |~ case1 "Indirect" int63_t (fun x -> Indirect x)
-      |~ case1 "Direct" H.t (fun x -> Direct x)
-      |> sealv
-
+    type dict_key = int [@@deriving irmin]
+    type pack_offset = int63 [@@deriving irmin]
+    type name = Indirect of dict_key | Direct of step
+    type address = Offset of pack_offset | Hash of H.t [@@deriving irmin]
```
Comment (Ngoguey42) [inline src/irmin-pack/inode.ml:306]: Are 
```suggestion
    type dict_key = int [@@deriving irmin]
    type pack_offset = int63 [@@deriving irmin]
    type name = Indirect of dict_key | Direct of step
    type address = Offset of pack_offset | Hash of H.t
```

**Review by Ngoguey42 — APPROVED** [review]
Thanks

File: src/irmin-pack/inode.ml (hunk 1 above)
Comment (CraigFe) [reply src/irmin-pack/inode.ml:306]: `address_t` is needed below in `ptr_t`, AFAICS

Comment (Ngoguey42) [reply src/irmin-pack/inode.ml:306]: Ah right, nevermind

**Pull request merged**